tests/core/contracts/test_implicit_contract.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from web3.contract import (
+    ImplicitContract,
+)
+
+
+@pytest.fixture()
+def math_contract(web3, MATH_ABI, MATH_CODE, MATH_RUNTIME):
+    # Deploy math contract
+    # NOTE Must use non-specialized contract factory or else deploy() doesn't work
+    MathContract = web3.eth.contract(
+        abi=MATH_ABI,
+        bytecode=MATH_CODE,
+        bytecode_runtime=MATH_RUNTIME,
+    )
+    tx_hash = MathContract.deploy()
+    tx_receipt = web3.eth.getTransactionReceipt(tx_hash)
+    math_address = tx_receipt['contractAddress']
+    # Return interactive contract instance at deployed address
+    # TODO Does parent class not implement 'deploy()' for a reason?
+    MathContract = web3.eth.contract(
+        abi=MATH_ABI,
+        bytecode=MATH_CODE,
+        bytecode_runtime=MATH_RUNTIME,
+        ContractFactoryClass=ImplicitContract,
+    )
+    return MathContract(math_address)
+
+
+@pytest.fixture()
+def get_transaction_count(web3):
+    def get_transaction_count(blocknum_or_label):
+        block = web3.eth.getBlock(blocknum_or_label)
+        # Return the blocknum if we requested this via labels
+        # so we can directly query the block next time (using the same API call)
+        # Either way, return the number of transactions in the given block
+        if blocknum_or_label in ["pending", "latest", "earliest"]:
+            return block.number, len(block.transactions)
+        else:
+            return len(block.transactions)
+    return get_transaction_count
+
+
+def test_implicitcontract_internals(math_contract):
+    # counter() is constant, so it calls by default
+    assert math_contract.counter.call_by_default
+    # return13() does not since it is not constant
+    assert not math_contract.return13.call_by_default
+
+
+def test_implicitcontract_call_default(math_contract, get_transaction_count):
+    blocknum, starting_txns = get_transaction_count("pending")
+    # When a function is called that defaults to call
+    # (Verify something was returned)
+    assert math_contract.counter() is not None
+    # Check that a call was made and not a transact
+    # (Auto-mining is enabled, so query by block number)
+    assert get_transaction_count(blocknum) == starting_txns
+
+
+def test_implicitcontract_transact_default(math_contract, get_transaction_count):
+    blocknum, starting_txns = get_transaction_count("pending")
+    # When a function is called that defaults to transact
+    # (Verify something was returned)
+    assert math_contract.return13() is not None
+    # Check that a transaction was made and not a call
+    # (Auto-mining is enabled, so query by block number)
+    assert get_transaction_count(blocknum) == starting_txns + 1
```

Comment:
I like it, will do

Reply:
I don't know `return13()` returns, but it certainly isn't the number 13 lol. I don't know where the source code is for this, what return value do you want me to check of this pre-compiled contract?

Reply:
Ah, I see how this is different. Disregard previous comment

Reply:
I covered all of these concerns, but not in every single test since 3/4 tests do transactions, which returns a txn_hash instead of the actual return value.